installers/npm/binary.js

```diff
@@ -136,6 +140,30 @@ const install = () => {
   }
 };
 
+const configureProxy = () => {
+  // get proxy env
+  const env = process.env.HTTP_PROXY || process.env.HTTPS_PROXY
```

Comment:
please use small case as well.

process.env.HTTP_PROXY || process.env.HTTPS_PROXY || process.env.http_proxy || process.env.https_proxy

Reply:
ideally, we should implement something like this - i.e check on to use http_proxy or https_proxy

https://github.com/request/request/blob/3c0cddc7c8eb60b470e9519da85896ed7ee0081e/lib/getProxyFromURI.js#L61-L66

Reply:
> ideally, we should implement something like this - i.e check on to use http_proxy or https_proxy
> 
> https://github.com/request/request/blob/3c0cddc7c8eb60b470e9519da85896ed7ee0081e/lib/getProxyFromURI.js#L61-L66

Good idea Kumar, do you know off hand where the URI param in this func is coming from, I could parse against the env if that is easier...

Reply:
https://github.com/request/request/blob/3c0cddc7c8eb60b470e9519da85896ed7ee0081e/lib/getProxyFromURI.js

you will need whole file which you can use as utility 

so, this is how proxy gets used
if the endpoint (which is being accessed) in no_proxy, if yes, then don't use proxy
if the endpoint (which is being accessed) is http url, then use http_proxy
if the endpoint (which is being accessed) is https url, then use https_proxy

(and to handle upper case of these)

Reply:
URI is the github.com URL form where tar is being downloaded

Reply:
I see what you mean, but I think due to this line:
https://github.com/farawaysouthwest/rover/blob/5f57ddd553ad581f1f551a85c124668a87d0b100/installers/npm/binary.js#L95

... the access url would always be HTTPS?

Reply:
Yes, in this case, it will be https_proxy for GitHub and would unblock us.

But these things depend on the organization. 

For ex:
What if if a organization has added GitHub in no_proxy. I.e don't use proxy for GitHub.com
As per current implementation, it will force the proxy and the connection might fail.

Hence, suggested having a generic solution.